src/server/models/search.ts

```diff
@@ -0,0 +1,26 @@
+import { QueryTypes } from 'sequelize'
+import { Url } from './url'
+import { sequelize } from '../util/sequelize'
+import { StorableUrlState } from '../repositories/enums'
+
+const { tableName } = Url
+const INDEX_NAME = 'urls_weighted_search_idx'
+
+// Warning: This expression has to be EXACTLY the same as the one used in the index
+// or else the index will not be used leading to unnecessarily long query times.
+export const urlSearchVector = `
+  setweight(to_tsvector('english', ${tableName}."shortUrl"), 'A') ||
+  setweight(to_tsvector('english', ${tableName}."description"), 'B')
+`
+
+export const urlSearchConditions = `urls.state = '${StorableUrlState.Active}' AND urls.description != ''`
+
+const dropIndex = `DROP INDEX IF EXISTS ${INDEX_NAME};`
+const createIndex = `CREATE INDEX urls_weighted_search_idx ON urls USING gin ((${urlSearchVector}))
```

Comment:
what happens when we deploy to multiple servers and the queries are interleaved? the DROP statement appears to be safe but will the CREATE INDEX statement fail if an index already exists?

Reply:
will using a BEGIN ... COMMIT help?

Reply:
Will add BEGIN, COMMIT. Will also be adding "IF NOT EXISTS" to the create statement. This way, at least one of the deployments should be able to commit with the new index.